[Feature-11440][Flink Stream] Use flink SDK replace the shell cli

shangeyao opened the pull request:
close #11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into dev

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
Merge remote-tracking branch 'upstream/dev' into feature-11440

shangeyao committed:
[Feature-11440][Flink Stream] Use flink SDK replace the shell cli

shangeyao committed:
[Feature-11440][Flink Stream] Use flink SDK replace the shell cli

shangeyao committed:
[Feature-11440][Flink Stream] Use flink SDK replace the shell cli

shangeyao committed:
[Feature-11440][Flink Stream] Use flink SDK replace the shell cli

caishunfeng committed:
fix workflow keep running when task fail (#11930)

tracehh committed:
[Feature][Api] Refactor&Add api.controller.QueueV2Controller (#11389)

* [Feature][Api] Refactor&Add api.controller.QueueV2Controller

* update review

shangeyao committed:
[Feature-11440][Flink Stream] fix code scanning error

shangeyao committed:
[Feature-11440][Flink Stream] fix code scanning error

DarkAssassinator committed:
[Bug] [Master] WorkflowExecuteRunnable will face a infinite loop #11838 (#11864)

* [Bug] [Master] WorkflowExecuteRunnable will face a infinite loop #11838

stalary committed:
[Bug][Dependent]: Id also clone due to duplicate when use dependent mode. (#11929)

EricGao888 committed:
[improve][Doc] Put default task parameters in a new file (#11776)

* Put default task parameters in a new file
* Add links to task plugin default parameters appendix docs
* Update Chinese docs

ruanwenjun committed:
Add debug logs for properties read from `common.properties` (#11910)

kezhenxu94 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-flink-stream/pom.xml:
```diff
@@ -15,19 +15,27 @@
 
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <flink.version>1.15.1</flink.version>
+        <kerb.version>2.0.1</kerb.version>
+        <log4j.version>2.18.0</log4j.version>
+        <sl4j.version>1.6.1</sl4j.version>
+        <flink.shaded.hadoop.version>2.8.3-7.0</flink.shaded.hadoop.version>
```
Don't manage the versions here, move them to the bom

kezhenxu94 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-flink-stream/src/main/java/org/apache/dolphinscheduler/plugin/task/flink/entity/CheckpointInfo.java:
```diff
@@ -0,0 +1,56 @@
+
+package org.apache.dolphinscheduler.plugin.task.flink.entity;
+
+public class CheckpointInfo {
```
Clean up this class with lombok annotations

kezhenxu94 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-flink-stream/src/main/java/org/apache/dolphinscheduler/plugin/task/flink/entity/ParamsInfo.java:
```diff
@@ -0,0 +1,359 @@
+import java.util.Arrays;
+import java.util.Properties;
+
+public class ParamsInfo {
```
Clean up this class with lombok annotations

kezhenxu94 commented on dolphinscheduler-task-plugin/dolphinscheduler-task-flink-stream/src/main/java/org/apache/dolphinscheduler/plugin/task/flink/entity/ResultInfo.java:
```diff
@@ -0,0 +1,36 @@
+
+    private String appId;
+    private String jobId;
+
+    public ResultInfo(String appId, String jobId) {
+        this.appId = appId;
+        this.jobId = jobId;
+    }
+
+    public ResultInfo() {
+
+    }
```
Clean up the CTORs with lombok annotations

shangeyao committed:
[Feature-11440][Flink Stream] use lombok && move dependency to bom

shangeyao committed:
[Feature-11440][Flink Stream] use lombok && move dependency to bom

shangeyao committed:
[Feature-11440][Flink Stream]move dependency to bom

shangeyao committed:
[Feature-11440][Flink Stream]move dependency to bom

itzhang89 committed:
[fix-10938]: use dot to replace source and make the default env work in shell (#11625)

* [fix-10938]: use the dot to replace source and make the default env work in the sh
Co-authored-by: Kerwin <37063904+zhuangchong@users.noreply.github.com>

kezhenxu94 committed:
Remove sensitive configs from dolphinscheduler_env.sh (#11917)

caishunfeng committed:
[Bug][Workflow Definition] fix ordinary users can not create depend task (#11961)

guodongym committed:
[Feature-10273][Datasource] Add dataSource for AWS Athena (#11157)

rickchengx committed:
[fix][doc] Update the registry related configuration in values.yaml (#11444)

ruanwenjun committed:
Fix alert_send_status may throw duplicate key exception and add limit for query alert (#11953)

kezhenxu94 committed:
Add validations of possible malicious keys (#11966)

jackfanwan committed:
Example Change the regular expression of the data source password (#11459)

ruanwenjun committed:
Fix cannot save processDefinition (#11931)

ruanwenjun committed:
Fix insertOrUpdate plugin may failed due to concurrent operation (#11471)

jieguangzhou committed:
[feat][python] Support Pytorch task in python api (#11975)

Co-authored-by: Jiajie Zhong <zhongjiajie955@gmail.com>

caishunfeng commented on dolphinscheduler-task-plugin/dolphinscheduler-task-flink-stream/src/main/java/org/apache/dolphinscheduler/plugin/task/flink/FlinkStreamTask.java:
```diff
@@ -55,37 +67,110 @@ public void init() {
+        String hadoopConfDir = System.getenv(HADOOP_CONF_DIR);
+
+        if (hadoopConfDir.isEmpty()) {
+            logger.error("Please make sure to set the HADOOP_CONF_DIR environment variable.");
```
Should throw taskException here?

shangeyao replied:
yeah, I correct it.

caishunfeng commented on dolphinscheduler-task-plugin/dolphinscheduler-task-flink-stream/src/main/java/org/apache/dolphinscheduler/plugin/task/flink/client/IClusterClient.java:
```diff
@@ -0,0 +1,167 @@
+import org.apache.dolphinscheduler.plugin.task.flink.enums.YarnTaskStatus;
+
+import java.util.List;
+public interface IClusterClient {
```
Please add some comment for it.

caishunfeng commented on dolphinscheduler-task-plugin/dolphinscheduler-task-flink-stream/src/main/java/org/apache/dolphinscheduler/plugin/task/flink/executor/AbstractClusterExecutor.java:
```diff
@@ -0,0 +1,300 @@
+                        "Deleting yarn application files under {} was unsuccessful.",
+                        applicationDir);
+            } else {
+                logger.error(
```
```suggestion
                logger.info(
```

shangeyao replied:
yeah, I correct it.

caishunfeng commented on dolphinscheduler-task-plugin/dolphinscheduler-task-flink-stream/src/main/java/org/apache/dolphinscheduler/plugin/task/flink/entity/ParamsInfo.java:
```diff
@@ -0,0 +1,62 @@
+@NoArgsConstructor
+@Builder
+@Data
+public class ParamsInfo {
```
```suggestion
public class FlinkParamsInfo {
```

shangeyao replied:
yeah, I correct it.

caishunfeng commented on dolphinscheduler-task-plugin/dolphinscheduler-task-flink-stream/src/main/java/org/apache/dolphinscheduler/plugin/task/flink/entity/ResultInfo.java:
```diff
@@ -0,0 +1,34 @@
+@NoArgsConstructor
+@Builder
+@Data
+public class ResultInfo {
```
```suggestion
public class FlinkResultInfo {
```

shangeyao replied:
yeah, I correct it.

caishunfeng commented on dolphinscheduler-task-plugin/dolphinscheduler-task-flink-stream/src/main/java/org/apache/dolphinscheduler/plugin/task/flink/executor/AbstractClusterExecutor.java:
```diff
@@ -0,0 +1,300 @@
+
+        } catch (Exception e) {
+            logger.error("flink job savepoint failed, appId:{}, jobId:{}, exception:{}", appId, jobId, e);
+            return new ResultInfo(appId, jobId);
```
```suggestion
            throw new TaskException("flink job savepoint failed", e)
```

shangeyao replied:
yeah, I correct it.

caishunfeng commented on dolphinscheduler-task-plugin/dolphinscheduler-task-flink-stream/src/main/java/org/apache/dolphinscheduler/plugin/task/flink/executor/AbstractClusterExecutor.java:
```diff
@@ -0,0 +1,300 @@
+                Object result = savepointFuture.get(2, TimeUnit.MINUTES);
+                logger.info("flink job savepoint path: {}", result.toString());
+            } catch (Exception e) {
+                logger.error("flink job savepoint error", e);
```
throw exception

shangeyao replied:
yeah, I correct it.

caishunfeng commented on dolphinscheduler-task-plugin/dolphinscheduler-task-flink-stream/src/main/java/org/apache/dolphinscheduler/plugin/task/flink/factory/YarnClusterDescriptorFactory.java:
```diff
@@ -0,0 +1,89 @@
+    @Override
+    public ClusterDescriptor createClusterDescriptor(
+                                                     String hadoopConfDir, Configuration flinkConfig) {
+        if (StringUtils.isNotBlank(hadoopConfDir)) {
+            try {
+                YarnConfiguration yarnConf = parseYarnConfFromConfDir(hadoopConfDir);
+                YarnClient yarnClient = createYarnClientFromYarnConf(yarnConf);
+
+                YarnClusterDescriptor clusterDescriptor =
+                        new YarnClusterDescriptor(
+                                flinkConfig,
+                                yarnConf,
+                                yarnClient,
+                                YarnClientYarnClusterInformationRetriever.create(yarnClient),
+                                false);
+                return clusterDescriptor;
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        } else {
+            throw new RuntimeException("yarn mode must set param of 'hadoopConfDir'");
+        }
+    }
```
```suggestion
        if (StringUtils.isBlank(hadoopConfDir)) {
            throw new RuntimeException("yarn mode must set param of 'hadoopConfDir'");
        }
          try {
              YarnConfiguration yarnConf = parseYarnConfFromConfDir(hadoopConfDir);
              YarnClient yarnClient = createYarnClientFromYarnConf(yarnConf);

              YarnClusterDescriptor clusterDescriptor =
                      new YarnClusterDescriptor(
                              flinkConfig,
                              yarnConf,
                              yarnClient,
                              YarnClientYarnClusterInformationRetriever.create(yarnClient),
                              false);
              return clusterDescriptor;
          } catch (Exception e) {
              throw new RuntimeException(e);
          }
```

shangeyao replied:
yeah, I correct it.

caishunfeng commented on dolphinscheduler-task-plugin/dolphinscheduler-task-flink-stream/src/test/java/org/apache/dolphinscheduler/plugin/task/flink/FlinkArgsUtilsTest.java:
```diff
@@ -17,116 +17,86 @@
 
 public class FlinkArgsUtilsTest {
 
-    private String joinStringListWithSpace(List<String> stringList) {
```
Maybe we should keep the old test for flink task, and add the new one for flink-stream task, WDYT?

shangeyao replied:
But I have changed some parameters, and the original UT is outdated. @caishunfeng

jieguangzhou committed:
[feat][python] Add sagemaker yaml example (#11925)

guowei-su committed:
[Improvement-11773][api] Optimize the log printing of the api module according… (#11782)

* [DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Resolve conflict.

* [DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- fix ProcessDefinitionServiceImpl.java dependency

* [DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Accept proposed change in ResourcePermissionCheckServiceImpl.java

* [DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Update the UT case in TenantServiceTest.java, UsersServiceTest.java, TaskGroupServiceTest.java.
- Update log printing in TaskGroupServiceImpl.java.
- Fix NPE of log parameter in ExecutorServiceImpl.java when call python api.

* [DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Fix NPE of log in WorkerGroupServiceImpl.java.
- Fix log parameter missing bug in ResourcesServiceImpl.java.

* [DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Fix bugs and handle some vulnerability codes.

* [DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Fix bugs and handle some vulnerability codes.

* [DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Fix e2e User test.

* [DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Fix codeQL check & SonarCloud Code Analysis.

* [DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Resolve import conflict.

* [DS-11773][api] Optimize the log printing of the api module according to the log specification doc.
- Fix codeQL check & SonarCloud Code Analysis.

EricGao888 committed:
[Doc][Style] Fix doc format once for all (#12006)

Zephyrcf committed:
[Feature-11473][Task]Support test task (#11670)

* [improve] support test tasks

* [improve] support test tasks

* [improve] support test tasks

* [improve] support test tasks

* [improve] support test tasks

* Update TaskExecuteThread.java

* try solve e2e q

* try solve e2e q

* try solve e2e q

* try solve e2e q

* try solve e2e q

* try solve e2e q

* try solve e2e q

* try solve e2e q

* try solve e2e q

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* Update DataSource.java

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* Update messages_zh_CN.properties

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* Update messages.properties

* Update messages_en_US.properties

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

* [Feature] support test tasks

jieguangzhou committed:
[feat][python] Support OpenMLDB task in python api (#11944)

Using the whole word zookeeper instead of short cut zk
Co-authored-by: Jiajie Zhong <zhongjiajie955@gmail.com>

jieguangzhou committed:
[Feature][PyDolphinScheduler] Support DVC task in pyds #11922 (#11941)

* add dvc task in pyds

* add BaseDVC class

mrlisir committed:
[Fix] Fixed an error importing project Maven (#12026)

jieguangzhou committed:
[feat][python] Support MLflow task in python api (#11962)

shangeyao committed:
[Feature-11440][Flink Stream]fix some bugs and add some exceptions

shangeyao committed:
[Feature-11440][Flink Stream]fix some bugs and add some exceptions

106umao committed:
and task instance date range search (#11963)

zhuangchong committed:
Fix link errors in release documentation. (#12049)

songjianet committed:
[Feat][UI] Update package. (#12046)

* [Feat][UI] Update package.

* [Feat][UI] Update package.

rickchengx committed:
[Improvement-11291][sql] Improvement the log output while executing multiple sql statements (#11292)

EricGao888 committed:
[Improvement][Test] Give an example on replacing Powermock with Mockito (#11588)

* Give an example on replacing powermock with mockito

* Remove redundant comments

* Refactoring related UTs for better readability.

jinyanhui2008 committed:
[Bug][ApiServer] When modifying the task definition, inserting the task definition log table will report the bug of primary key conflict #12037

Co-authored-by: 金鑫 <jinyanhui@huansi.net>

Zephyrcf committed:
[Improve] Fix support test tasks UT (#12062)

jackfanwan committed:
[improvement]Add Set cluster name (#12058)

* Add Set cluster name

* add unit test

Co-authored-by: fanwanlong <fanwanlong@kezaihui.com>

ruanwenjun committed:
Fix update TaskDefinition error (#12060)

EricGao888 committed:
[Feature][Resource Center] Add support for Alibaba Cloud OSS as storage of resource center (#11708)

* Add support for Alibaba Cloud OSS as storage of resource center

* Add dependency licenses

* Refactor code, add unit tests and related docs

ruanwenjun committed:
Script cannot contains ''' in params (#12067)

limaiwang committed:
add(dependent node's add unprivileged project list): dependent node's add unprivileged project list (#12073)

jackfanwan committed:
[Improvement] Add Set cluster name (#12074)

EricGao888 committed:
[Improvement][UT] Upgrade jUnit to 5.+ (#10976) (#11332)

* [Improvement][UT] Upgrade jUnit to 5.+ (#10976)

* Refactor AlertServerTest with jUnit-5 as an example

jieguangzhou committed:
[improve] Optimize MLFlow task plugin for easy of use (#12071)

optimize code

rickchengx committed:
[improve] Remove the spark version of spark task (#11860)

fengjian1129 committed:
[Bug-#12057][task-plugin] fix when the sql query result is empty, the email fails to send the attachment, and an exception will be reported (#12059)

* [Bug-#12057][task-plugin] fix when the sql query result is empty, the email fails to send the attachment, and an exception will be reported

Co-authored-by: 冯剑 Jian <jian.feng@jiduauto.com>

ruanwenjun committed:
Fix insert command error due to the id is not null (#12092)

Commit:
[Feature] Add CURD to the project/tenant/user section of the python-DS (#11162)

- Add CURD in project
- Add CURD in tenant
- Add CURD in user
- Add test in user

Co-authored-by: Jiajie Zhong <zhongjiajie955@gmail.com>

jackfanwan committed:
[BUG]Delete Setting ID (#12096)

* Delete Setting ID

* modify id

Co-authored-by: jackfanwan <fanwanlong@kezaihui.com>

xdu-chenrj committed:
[feat][python] new resource plugin GitHub and gitlab (#11831)

Co-authored-by: Jiajie Zhong <zhongjiajie955@gmail.com>
Co-authored-by: JieguangZhou <jieguang_zhou@163.com>
Co-authored-by: Devosend <devosend@gmail.com>
Co-authored-by: juzimao <578961953@qq.com>

Amy0104 committed:
[Improvement][UI] Add filterable for `src_table` and `target_table` in the data quality task. (#12099)

jieguangzhou committed:
fix python home conflict (#12112)

zhongjiajie committed:
[Fix] Also auto formatter workflow instance if location is null (#12080)

* [fix] Also auto formatter workflow instance if location is null

In #11535(096fae77) and #11681(4dca488c), we already formatter
workflow definition, but I find out we forget formatter workflow
instance with definition's location is null, this patch also
formatter workflow instance.

* remove console log

zhongjiajie committed:
[fix][python] Task switch branch not show in webui (#12120)

when create task switch in python api, switch branch not show correctly,
due to we add some unnecessary attribute to switch task, this patch we
add `_task_ignore_attr` in class `Task` to ignore those unnecessary
attribute from `_task_default_attr`

zhongjiajie committed:
[Feature][API] New restful API for workflow and schedule  (#11912)

* [feat] New restful API for workflow and schedule

CURD for workflow and schedule, different with exists
API, this new restful api only operate single resource
in each request, and return the latest. For example,
previous workflow should also need to post tasks definition
and tasks relation definition, but this patch will allow
you to create workflow without task relate information

* use checkProjectAndAuthThrowException, and fix CI error

* Update dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ScheduleV2Controller.java

* change method name
from createProcessDefinitionV2 to createSingleProcessDefinition
from updateProcessDefinitionV2 to updateSingleProcessDefinition

Co-authored-by: caishunfeng <caishunfeng2021@gmail.com>

EricGao888 committed:
[Doc][Improvement] Add instructions for process execution type and data complement (#12121)

* Add instructions for process execution type and data complement

* Fix and polish the docs

DarkAssassinator committed:
[fix#12000]Cannot remove the WorkerGroup from the master service (#12050)

* [Bug] [Master] Cannot remove the WorkerGroup from the master service. #12000

* remove unnecessary locks

* Update dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/registry/ServerNodeManager.java

Co-authored-by: caishunfeng <caishunfeng2021@gmail.com>

Co-authored-by: caishunfeng <caishunfeng2021@gmail.com>

DarkAssassinator committed:
[Bug] [Common] update IPv4 Pattern (#11762)

* [Bug] [Common] IPv4 Pattern is wrong. #11758

labbomb committed:
refactor: dag-menu refactor (#11965)

* refactor: dag-menu refactor

* feat: add i18n support for the dag-menu

* feat: add e2e class

Co-authored-by: devosend <devosend@gmail.com>
Co-authored-by: Tq <tianqitobethefirst@gmail.com>

zhuxt2015 committed:
[Improvement][SQL Task]use default sql segment separator (#10869)

Tianqi-Dotes committed:
[BUG][BE-12113]fix htrace dependency conflicts (#12126)

* fix htrace dependency conflicts

* replace htrace-core with hbase-noop-htrace

caishunfeng committed:
fix upgrade sql (#12141)

fengjian1129 committed:
[Improvement-#11608][task-plugin] New time variables are added to facilitate business development (#11667)

Add new time expressions for users' convenience

Co-authored-by: 冯剑 Jian <jian.feng@jiduauto.com>
Co-authored-by: David <dailidong66@gmail.com>

zhuangchong committed:
[release][doc] Add release 3.0.1 (#12139)

* add release 3.0.1
* add 30x version in the github bug report

EricGao888 committed:
Remove powermock in dolphinscheduler-master module (#12143)

* Remove powermock related code in dolphinscheduler-master module
* Remove redundant comments, useless code and unnecessary stubbings

devosend committed:
fix env config space bug (#12147)

zhongjiajie committed:
[fix][doc] Do not change docsdev.js during releasing (#12151)

ref: https://github.com/apache/dolphinscheduler/pull/12146#discussion_r979708629
and https://github.com/apache/dolphinscheduler-website/pull/829

tracehh committed:
[Improvement][Master] Construct processInstance may NPE when master handling command (#12056)

* [Improvement][Master] Construct processInstance may NPE when master handling command

* use an enpty map

Co-authored-by: xuhaihui <xuhaihui@cmss.chinamobile.com>

shangeyao committed:
add dependency license
upgrade flink version to 1.15.2

shangeyao committed:
add dependency license
upgrade flink version to 1.15.2

shangeyao committed:
resolve conflicts

shangeyao committed:
Merge remote-tracking branch 'origin/feature-11440' into feature-11440

# Conflicts:
#	dolphinscheduler-bom/pom.xml
#	dolphinscheduler-task-plugin/dolphinscheduler-task-flink-stream/src/test/java/org/apache/dolphinscheduler/plugin/task/flink/FlinkParamBuilderTest.java
#	tools/dependencies/known-dependencies.txt

EricGao888 commented:
Could u plz fix the commit history of this PR? Thanks.
